File: rust/src/decoding/double_decoder.rs
use crate::decoding::decoding_utils::DecodingUtils;
use crate::headers::stream::StreamMetadata;

pub struct DoubleDecoder {}
impl DoubleDecoder {
    pub fn decode_double_stream(data: &[u8], offset: &mut usize, stream_metadata: &StreamMetadata) -> Vec<f64> {
        DecodingUtils::decode_doubles_le(data, offset, stream_metadata.num_values as usize)
    }
}

File: ts/src/encodings/doubleDecoder.ts
import type IntWrapper from './intWrapper';
import { type StreamMetadata } from '../metadata/tile/streamMetadata';
import {decodeDoublesLE, decodeFloatsLE} from "./decodingUtils";

export function decodeFloatStream(data: Uint8Array, offset: IntWrapper, streamMetadata: StreamMetadata): Float32Array {
        const values = decodeFloatsLE(data, offset, streamMetadata.numValues);
        const valuesList: Float32Array = new Float32Array(values.length);
        for (let i = 0; i < values.length; i++) {
            valuesList[i] = values[i];
        }
        return valuesList;
}

export function decodeDoubleStream(data: Uint8Array, offset: IntWrapper, streamMetadata: StreamMetadata): Float32Array {
    const values = decodeDoublesLE(data, offset, streamMetadata.numValues);
    const valuesList: Float32Array = new Float32Array(values.length);
    for (let i = 0; i < values.length; i++) {
        valuesList[i] = values[i];
    }
    return valuesList;
}
